lib/phoenix/channel/transport.ex

```diff
@@ -112,65 +118,61 @@ defmodule Phoenix.Channel.Transport do
   The server will respond to heartbeats with the same message
   """
   def dispatch(socket, "phoenix", "heartbeat", _msg) do
-    send socket.pid, {:socket_reply, %Message{topic: "phoenix", event: "heartbeat", payload: %{}}}
+    socket
+    |> Socket.Server.dispatch_reply(%Message{topic: "phoenix", event: "heartbeat", payload: %{}})
```

Comment:
It feels weird to send this request to the server just so it sends it back to us. I think the heartbeat is a transport heartbeat and not channel one (so we can skip going through the channel directly).